packages/omgidl-serialization/src/MessageReader.test.ts

```diff
@@ -1326,6 +1326,20 @@ module builtin_interfaces {
     `;
     const data = {};
     const writer = new CdrWriter({ kind: EncapsulationKind.PL_CDR_LE });
+
+    const rootDef = "Message";
+    const reader = new MessageReader(rootDef, parseIDL(msgDef));
+    const msgout = reader.readMessage(writer.data);
+    expect(msgout).toEqual(data);
+  });
+  it("Reads mutable struct with no members", () => {
+    const msgDef = `
+      @mutable
+      struct Message {
+      };
+    `;
+    const data = {};
+    const writer = new CdrWriter({ kind: EncapsulationKind.PL_CDR_LE });
```

Comment:
Since you only wrote a sentinel header it's considering this struct as "not present". It would be considered populated and present if an emHeader was written for it. Should probably have a test for both of these cases. Sorry for repeated goal-post moving 😅 but it would be good to have